src/main/java/uk/nhs/digital/nhsconnect/nhais/outbound/AbstractToEdifactService.java

```diff
@@ -36,6 +38,7 @@ protected OutboundMeshMessage convert(T translationItems) {
         Objects.requireNonNull(translationItems.getRecipient(), "Recipient can't be null");
         Objects.requireNonNull(translationItems.getTransactionType(), "Transaction type can't be null");
 
+        validateRecipient(translationItems);
```

Comment:
Why just check the recipient here? Do the real lookup and include it in the published message. Our Mesh classes shoudn't know anything about edifact / nhais / gp links.

Reply:
This is fine as is